crates/runtime/src/forge_runtime_extension.rs

```diff
@@ -448,6 +439,140 @@ impl<'a> ExtensionLogic
                     ]))
                 }
             }
+            "get_stark_public_key" => {
+                let private_key = inputs[0].clone();
+                let key_pair = SigningKey::from_secret_scalar(private_key.into_());
+
+                Ok(CheatcodeHandlingResult::Handled(vec![key_pair
+                    .verifying_key()
+                    .scalar()
+                    .into_()]))
+            }
+            "generate_ecdsa_keys" => {
+                let curve = inputs[0].clone().to_u8();
+
+                let (signing_key_bytes, verifying_key_bytes) = {
+                    match curve {
+                        Some(0) => {
+                            let signing_key = k256::ecdsa::SigningKey::random(
+                                &mut k256::elliptic_curve::rand_core::OsRng,
+                            );
+                            let verifying_key = signing_key.verifying_key();
+                            (
+                                signing_key.to_bytes(),
+                                verifying_key.to_encoded_point(false).to_bytes(),
+                            )
+                        }
+                        Some(1) => {
+                            let signing_key = p256::ecdsa::SigningKey::random(
+                                &mut p256::elliptic_curve::rand_core::OsRng,
+                            );
+                            let verifying_key = signing_key.verifying_key();
+                            (
+                                signing_key.to_bytes(),
+                                verifying_key.to_encoded_point(false).to_bytes(),
+                            )
+                        }
+                        _ => unreachable!("Invalid EllipticCurve variant"),
+                    }
+                };
+
+                Ok(CheatcodeHandlingResult::Handled(vec![
+                    Felt252::from_bytes_be(&signing_key_bytes[16..32]),
+                    Felt252::from_bytes_be(&signing_key_bytes[0..16]),
+                    Felt252::from_bytes_be(&verifying_key_bytes[17..33]),
+                    Felt252::from_bytes_be(&verifying_key_bytes[1..17]),
+                    Felt252::from_bytes_be(&verifying_key_bytes[49..65]),
+                    Felt252::from_bytes_be(&verifying_key_bytes[33..49]),
+                ]))
```

Comment:
These come from the different order of halves in cairo u256

Reply:
The order of bytes could be changed here to "normal", but then it would be reversed in cairo.